testing/test_optimizers.py

```diff
@@ -2,6 +2,14 @@
 import unittest
 import numpy as np
 import GPflow
+from contextlib import contextmanager
+from scipy.optimize import OptimizeResult
+
+import sys
+if sys.version_info[0] == 3:
+    from io import StringIO
+else:
+    from io import BytesIO as StringIO
```

Comment:
As long as this is in the testing framework it is indeed OK I think. I'd like to avoid python 2 vs 3 code in the core, at least when used without some easy compatibility lib like Six.